src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/abstractjavadoc/InputAbstractJavadocNonTightHtmlTags2.java

```diff
@@ -19,10 +19,10 @@
  */
 // violation 9 lines above 'tag BODY_TAG_START'
 // violation 9 lines above 'tag P_TAG_START'
-// 2 violations 8 lines above
+// 2 violations 8 lines above:
```

Comment:
Please share how we fail on +/-1, i.e. `1 violations 8 lines above:` and `3 violations 8 lines above:`.

Reply:
```
✘-1 ~/java/github/romani/checkstyle [i13307-miltiline L|✚ 1] 
05:25 $ git diff
diff --git a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/abstractjavadoc/InputAbstractJavadocNonTightHtmlTags2.java b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/abstractjavadoc/InputAbstractJavadocNonTightHtmlTags2.java
index 0df01f3..71f5015 100644
--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/abstractjavadoc/InputAbstractJavadocNonTightHtmlTags2.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/abstractjavadoc/InputAbstractJavadocNonTightHtmlTags2.java
@@ -19,7 +19,7 @@ package com.puppycrawl.tools.checkstyle.checks.javadoc.abstractjavadoc;
  */
 // violation 9 lines above 'tag BODY_TAG_START'
 // violation 9 lines above 'tag P_TAG_START'
-// 2 violations 8 lines above:
+// 1 violations 8 lines above:
 //                            'Unclosed HTML tag found: p'
 //                            'tag P_TAG_START'
 // 2 violations 10 lines above:
```
results:
```
[INFO] Running com.puppycrawl.tools.checkstyle.checks.javadoc.AbstractJavadocCheckTest
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 0.986 s <<< FAILURE! -- in com.puppycrawl.tools.checkstyle.checks.javadoc.AbstractJavadocCheckTest
[ERROR] com.puppycrawl.tools.checkstyle.checks.javadoc.AbstractJavadocCheckTest.testVisitCountForCheckAcceptingJavadocWithNonTightHtml -- Time elapsed: 0.962 s <<< FAILURE!
Violation lines for /home/rivanov/java/github/romani/checkstyle/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/abstractjavadoc/InputAbstractJavadocNonTightHtmlTags2.java differ.
unexpected (1): 14
---
expected      : [11, 12, 14, 15, 15, 30, 30, 30, 37, 37, 44, 45, 45, 45, 55, 55, 56, 65, 66, 67, 67, 67, 78, 78, 78, 80, 80, 80, 96, 98, 98, 107, 108, 111]
but was       : [11, 12, 14, 14, 15, 15, 30, 30, 30, 37, 37, 44, 45, 45, 45, 55, 55, 56, 65, 66, 67, 67, 67, 78, 78, 78, 80, 80, 80, 96, 98, 98, 107, 108, 111]
	at com.puppycrawl.tools.checkstyle.AbstractModuleTestSupport.verifyViolations(AbstractModuleTestSupport.java:470)
	at com.puppycrawl.tools.checkstyle.AbstractModuleTestSupport.verifyWithInlineConfigParser(AbstractModuleTestSupport.java:238)
	at com.puppycrawl.tools.checkstyle.checks.javadoc.AbstractJavadocCheckTest.testVisitCountForCheckAcceptingJavadocWithNonTightHtml(AbstractJavadocCheckTest.java:470)
	at [[Reflective call: 4 frames collapsed (https://goo.gl/aH3UyP)]].(:0)
	at [[Testing framework: 27 frames collapsed (https://goo.gl/aH3UyP)]].(:0)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
	at [[Testing framework: 9 frames collapsed (https://goo.gl/aH3UyP)]].(:0)
	at java.base/java.util.ArrayList.forEach(ArrayList.java:1541)
	at [[Testing framework: 21 frames collapsed (https://goo.gl/aH3UyP)]].(:0)
	at org.apache.maven.surefire.junitplatform.LazyLauncher.execute(LazyLauncher.java:56)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.execute(JUnitPlatformProvider.java:184)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invokeAllTests(JUnitPlatformProvider.java:148)
	at org.apache.maven.surefire.junitplatform.JUnitPlatformProvider.invoke(JUnitPlatformProvider.java:122)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:385)
	at org.apache.maven.surefire.booter.ForkedBooter.execute(ForkedBooter.java:162)
	at org.apache.maven.surefire.booter.ForkedBooter.run(ForkedBooter.java:507)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:495)

[INFO] 
[INFO] Results:
[INFO] 
[ERROR] Failures: 
[ERROR]   AbstractJavadocCheckTest.testVisitCountForCheckAcceptingJavadocWithNonTightHtml:470->AbstractModuleTestSupport.verifyWithInlineConfigParser:238->AbstractModuleTestSupport.verifyViolations:470 Violation lines for /home/rivanov/java/github/romani/checkstyle/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/abstractjavadoc/InputAbstractJavadocNonTightHtmlTags2.java differ.
unexpected (1): 14
---
expected      : [11, 12, 14, 15, 15, 30, 30, 30, 37, 37, 44, 45, 45, 45, 55, 55, 56, 65, 66, 67, 67, 67, 78, 78, 78, 80, 80, 80, 96, 98, 98, 107, 108, 111]
but was       : [11, 12, 14, 14, 15, 15, 30, 30, 30, 37, 37, 44, 45, 45, 45, 55, 55, 56, 65, 66, 67, 67, 67, 78, 78, 78, 80, 80, 80, 96, 98, 98, 107, 108, 111]
[INFO] 
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0
[INFO] 
```

Reply:
missed message resulted in no error, so extra fix is required:
```
 ~/java/github/romani/checkstyle [i13307-miltiline L|✚ 2] 
$ git diff
diff --git a/src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java b/src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java
index 229dc33..577ddf6 100644
--- a/src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java
+++ b/src/test/java/com/puppycrawl/tools/checkstyle/bdd/InlineConfigParser.java
@@ -606,6 +606,11 @@ public final class InlineConfigParser {
                 final String violationMessage = messageMatcher.group(1);
                 result.add(new TestInputViolation(violationLineNum, violationMessage));
             }
+            else {
+                throw new IllegalStateException("At least one message is missed"
+                    + " under " + lineNo + " for line "
+                    + violationLineNum);
+            }
         }
         return result;
     }
diff --git a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/abstractjavadoc/InputAbstractJavadocNonTightHtmlTags2.java b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/abstractjavadoc/InputAbstractJavadocNonTightHtmlTags2.java
index 0df01f3..dd9b59c 100644
--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/abstractjavadoc/InputAbstractJavadocNonTightHtmlTags2.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/abstractjavadoc/InputAbstractJavadocNonTightHtmlTags2.java
@@ -19,7 +19,7 @@ package com.puppycrawl.tools.checkstyle.checks.javadoc.abstractjavadoc;
  */
 // violation 9 lines above 'tag BODY_TAG_START'
 // violation 9 lines above 'tag P_TAG_START'
-// 2 violations 8 lines above:
+// 3 violations 8 lines above:
 //                            'Unclosed HTML tag found: p'
 //                            'tag P_TAG_START'
 // 2 violations 10 lines above:
```

results:
```
[INFO] Running com.puppycrawl.tools.checkstyle.checks.javadoc.AbstractJavadocCheckTest
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.147 s <<< FAILURE! -- in com.puppycrawl.tools.checkstyle.checks.javadoc.AbstractJavadocCheckTest
[ERROR] com.puppycrawl.tools.checkstyle.checks.javadoc.AbstractJavadocCheckTest.testVisitCountForCheckAcceptingJavadocWithNonTightHtml -- Time elapsed: 0.120 s <<< ERROR!
java.lang.IllegalStateException: at least one message is missed under 21 for line 14
	at com.puppycrawl.tools.checkstyle.bdd.InlineConfigParser.getMultipleViolations(InlineConfigParser.java:609)
	at com.puppycrawl.tools.checkstyle.bdd.InlineConfigParser.setViolations(InlineConfigParser.java:562)
	at com.puppycrawl.tools.checkstyle.bdd.InlineConfigParser.setViolations(InlineConfigParser.java:458)
	at com.puppycrawl.tools.checkstyle.bdd.InlineConfigParser.parse(InlineConfigParser.java:196)
	at com.puppycrawl.tools.checkstyle.bdd.InlineConfigParser.parse(InlineConfigParser.java:172)
	at com.puppycrawl.tools.checkstyle.AbstractModuleTestSupport.verifyWithInlineConfigParser(AbstractModuleTestSupport.java:235)
	at com.puppycrawl.tools.checkstyle.checks.javadoc.AbstractJavadocCheckTest.testVisitCountForCheckAcceptingJavadocWithNonTightHtml(AbstractJavadocCheckTest.java:470)
....
```

Reply:
Since we all have details in this method can we do something like: 

```
Expected number of violation messages does not match actual number of violation messages on line '21'.
Expected: 3
Actual: 2
```

Reply:
we can not assert by Thruth as is it is parser not a test.
but message is updated:
```
diff --git a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/abstractjavadoc/InputAbstractJavadocNonTightHtmlTags2.java b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/abstractjavadoc/InputAbstractJavadocNonTightHtmlTags2.java
index 0df01f3..dd9b59c 100644
--- a/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/abstractjavadoc/InputAbstractJavadocNonTightHtmlTags2.java
+++ b/src/test/resources/com/puppycrawl/tools/checkstyle/checks/javadoc/abstractjavadoc/InputAbstractJavadocNonTightHtmlTags2.java
@@ -19,7 +19,7 @@ package com.puppycrawl.tools.checkstyle.checks.javadoc.abstractjavadoc;
  */
 // violation 9 lines above 'tag BODY_TAG_START'
 // violation 9 lines above 'tag P_TAG_START'
-// 2 violations 8 lines above:
+// 3 violations 8 lines above:
 //                            'Unclosed HTML tag found: p'
 //                            'tag P_TAG_START'
 // 2 violations 10 lines above:


[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running com.puppycrawl.tools.checkstyle.checks.javadoc.AbstractJavadocCheckTest
[ERROR] Tests run: 1, Failures: 0, Errors: 1, Skipped: 0, Time elapsed: 0.134 s <<< FAILURE! -- in com.puppycrawl.tools.checkstyle.checks.javadoc.AbstractJavadocCheckTest
[ERROR] com.puppycrawl.tools.checkstyle.checks.javadoc.AbstractJavadocCheckTest.testVisitCountForCheckAcceptingJavadocWithNonTightHtml -- Time elapsed: 0.108 s <<< ERROR!
java.lang.IllegalStateException: Declared amount of violation messages at line 22 is 3 but found 2
  at com.puppycrawl.tools.checkstyle.bdd.InlineConfigParser.getMultipleViolations(InlineConfigParser.java:621)
  at com.puppycrawl.tools.checkstyle.bdd.InlineConfigParser.setViolations(InlineConfigParser.java:567)
  at com.puppycrawl.tools.checkstyle.bdd.InlineConfigParser.setViolations(InlineConfigParser.java:463)
  at com.puppycrawl.tools.checkstyle.bdd.InlineConfigParser.parse(InlineConfigParser.java:201)
  at com.puppycrawl.tools.checkstyle.bdd.InlineConfigParser.parse(InlineConfigParser.java:177)
```